libsignal-protocol/src/crypto/openssl.rs

```diff
@@ -68,14 +70,18 @@ impl Crypto for OpenSSLCrypto {
 
     fn hmac_sha256(
         &self,
-        _key: &[u8],
+        key: &[u8],
     ) -> Result<Box<dyn Sha256Hmac>, InternalError> {
-        let nid = Nid::HMACWITHSHA256;
-        let ty = MessageDigest::from_nid(nid)
-            .ok_or_else(|| InternalError::Unknown)?;
-        let hasher = Hasher::new(ty).map_err(|_e| InternalError::Unknown)?;
-
-        Ok(Box::new(hasher))
+        let pkey =
+            Box::new(PKey::hmac(key).map_err(|_e| InternalError::Unknown)?);
+        let signer = {
+            // a little bit a hack here, but i think it's safe, since the
+            // `Sha256Hmac` is static anyway !
```

Comment:
Nope, it's not static. We're creating a *new* `Sha256Hmac` here every time it's called, not returning a pointer to a static variable.

Reply:
Oh, i missed that part, i will try to refactor it and use `rental`